include/gz/sim/ServerConfig.hh

```diff
@@ -386,7 +386,19 @@ namespace gz
                   const std::string &_apiBackend);
 
       /// \return Api backend for gui. See SetRenderEngineGuiApiBackend()
-      const std::string &RenderEngineGuiApiBackend() const;
+      public: const std::string &RenderEngineGuiApiBackend() const;
+
+      /// \brief Set whether the server to should stop loading in presence
+      /// of SDF errors.
+      /// \param[in] _blockOnSdfErrors True to stop loading the server in
+      /// presence of SDF errors, false to continue loading if SDF errors exist.
+      public: void SetBlockOnSdfErrors(bool _blockOnSdfErrors);
```

Comment:
This would be more extendable if we used an enum here (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#i4-make-interfaces-precisely-and-strongly-typed). So we'd have `SetBehaviorOnSdfErrors(SdfErrorBehavior _behavior)` and for now, we would just have 
```cpp
enum class SdfErrorBehavior
{
  EXIT_IMMEDIATELY
}
```

Reply:
changed to use enum. I also added `CONTINUE_LOADING` enum value so that we can tell server to continue loading in presence of SDF errors. [afd6845](https://github.com/gazebosim/gz-sim/pull/2349/commits/afd6845cee60b36e0581fdaa99264dcc36b64438)